service/src/test/java/com/ge/predix/controller/test/AttributeConnectorControllerIT.java

```diff
@@ -0,0 +1,130 @@
+
+    private MockMvc mockMvc;
+
+    @BeforeClass
+    public void setup() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).build();
+    }
+
+    @Test
+    public void testCreateAndGetAndDeleteResourceConnector() throws Exception {
+        createZone1AndAssert();
+
+        AttributeConnector resourceConfig = this.jsonUtils
+                .deserializeFromFile("controller-test/createResourceConnector.json", AttributeConnector.class);
+        Assert.assertNotNull(resourceConfig, "createResourceConnector.json file not found or invalid");
+        String resourceConfigContent = this.objectWriter.writeValueAsString(resourceConfig);
+        this.mockMvc.perform(put(V1_RESOURCE_CONNECTOR_URL).contentType(MediaType.APPLICATION_JSON)
+                .content(resourceConfigContent)).andExpect(status().isCreated());
+
+        this.mockMvc.perform(get(V1_RESOURCE_CONNECTOR_URL))
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+                // .andExpect(jsonPath("isActive", is("true")))
+                .andExpect(jsonPath("maxCachedIntervalMinutes", is(60)))
+                .andExpect(jsonPath("adapters[0].adapterEndpoint", is("https://my-adapter.com")))
+                .andExpect(jsonPath("adapters[0].uaaTokenUrl", is("https://my-uaa.com")))
+                .andExpect(jsonPath("adapters[0].uaaClientId", is("adapter-client")))
+                // .andExpect(jsonPath("adapters[0].uaaClientSecret", is("adapter-secret")))
+                .andExpect(status().isOk());
+
+        this.mockMvc.perform(delete(V1_RESOURCE_CONNECTOR_URL)).andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void testGetResourceConnectorWhichDoesNotExists() throws Exception {
+        createZone1AndAssert();
+        this.mockMvc.perform(get(V1_RESOURCE_CONNECTOR_URL)).andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void testDeleteResourceConnectorWhichDoesNotExist() throws Exception {
+        createZone1AndAssert();
+        this.mockMvc.perform(delete(V1_RESOURCE_CONNECTOR_URL)).andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void testCreateResourceConnectorWithEmptyAdapters() throws Exception {
+        createZone1AndAssert();
+        AttributeConnector connector = this.jsonUtils.deserializeFromFile(
+                "controller-test/createResourceConnectorWithEmptyAdapters.json", AttributeConnector.class);
+        Assert.assertNotNull(connector, "createResourceConnectorWithEmptyAdapters.json file not found or invalid");
+        String connectorContent = this.objectWriter.writeValueAsString(connector);
+
+        this.mockMvc.perform(put(V1_RESOURCE_CONNECTOR_URL).contentType(MediaType.APPLICATION_JSON)
+                .content(connectorContent)).andExpect(status().isUnprocessableEntity());
+    }
+
+    @Test
+    public void testCreateResourceConnectorWithCacheaIntervalBelowThreshold() throws Exception {
```

Comment:
the validation tests here should suffice, if we use jackson for validation. Please remove any similar validation tests from the unit test - AttributeConnectorServiceTest